File: core/src/org/rexcellentgames/burningknight/debug/GiveCommand.java
package org.rexcellentgames.burningknight.debug;

import org.rexcellentgames.burningknight.entity.creature.player.Player;
import org.rexcellentgames.burningknight.entity.item.Item;
import org.rexcellentgames.burningknight.entity.item.ItemHolder;
import org.rexcellentgames.burningknight.entity.item.ItemRegistry;

public class GiveCommand extends ConsoleCommand {
	{
		shortName = "/gv";
		name = "/give";
	}

	@Override
	public void run(Console console, String[] args) {
		int count = 1;

		if (args.length == 2) {
			count = Integer.valueOf(args[1]);
		}

		if (args.length > 0 && args.length < 3) {
			String name = args[0];
			Item item;

			try {
				ItemRegistry.Pair clazz = ItemRegistry.INSTANCE.getItems().get(name);

				if (clazz == null) {
					console.print("[red]Unknown item $name");
					return;
				}

				item = clazz.getType().newInstance();

				if (item.isStackable()) {
					item.setCount(count);
				}

				ItemHolder itemHolder = new ItemHolder();
				itemHolder.setItem(item);

				Player.instance.tryToPickup(itemHolder);
				console.print("[green]Gave " + name + " (" + count + ")");
			} catch (Exception e) {
				console.print("[red]Failed to create item");
				e.printStackTrace();
			}
		}
	}
}

File: core/src/org/rexcellentgames/burningknight/debug/GiveCommand.kt
package org.rexcellentgames.burningknight.debug

import org.rexcellentgames.burningknight.entity.creature.player.Player
import org.rexcellentgames.burningknight.entity.item.Item
import org.rexcellentgames.burningknight.entity.item.ItemHolder
import org.rexcellentgames.burningknight.entity.item.ItemRegistry
import org.rexcellentgames.burningknight.util.Log

class GiveCommand : ConsoleCommand("/give", "/gv", "[item] (count) gives an item") {
  override fun run(console: Console, args: Array<String>) {
    var count = 1

    if (args.size == 2) {
      count = Integer.valueOf(args[1])
    }

    if (args.size > 0 && args.size < 3) {
      val name = args[0]
      val item: Item
      
      try {
        item = if (name.contains(":")) {
          ItemRegistry.modItems[name] as Item
        } else {
          val clazz = ItemRegistry.items[name]

          if (clazz == null) {
            Log.error("Unknown item $name")

            return
          }

          clazz.type.newInstance()
        }

        if (item.isStackable) {
          item.count = count
        }
        
        val itemHolder = ItemHolder()
        itemHolder.item = item
        
        Player.instance.tryToPickup(itemHolder)
      } catch (e: Exception) {
        Log.error("Failed to create item, consult @egordorichev")
        e.printStackTrace()
      }
    }
  }
}